### PR TITLE
Fix missing entrypoints and add build-time validation

### LINKED ISSUE #54784: Entrypoint verification should work on Apple platforms
If you add a new entry point in the unix shims, but forget to add it to `entrypoints.c`, you get a nice build error:

```
EXEC : error : /workspaces/runtime/src/libraries/Native/Unix/System.Security.Cryptography.Native/entrypoints.c file did not match entries exported from /workspaces/runtime/artifacts/obj/native/net6.0-Linux-Debug-x64/System.Security.Cryptography.Native/libSystem.Security.Cryptography.Native.OpenSsl.so [/workspaces/runtime/src/libraries/Native/build-native.proj]
  DIFFERENCES FOUND: 
  CryptoNative_Foob
  ninja: build stopped: subcommand failed.
```

This does not appear to work for macOS. Adding a new exported function and omitting it from `entrypoints.c` does not fail the build.

### PR DESCRIPTION
**Pull request by vcsjones**
This adds a bunch of missing `DllImportEntry`s and brings in build-time validation to prevent any additional drift.

Closes #54784 

/cc @VSadov

### CODE REVIEW

**Comment by vcsjones** [comment]
Example output when I added an exported function without including it in the entry point list:


```
EXEC : error : /Users/kjones/Projects/dotnet/runtime-main/src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/../../Unix/System.Security.Cryptography.Native.Apple/entrypoints.c file did not match entries exported from /Users/kjones/Projects/dotnet/runtime-main/artifacts/obj/native/net6.0-OSX-Debug-x64/System.Security.Cryptography.Native.Apple/libSystem.Security.Cryptography.Native.Apple.dylib [/Users/kjones/Projects/dotnet/runtime-main/src/libraries/Native/build-native.proj]
  DIFFERENCES FOUND: 
  AppleCryptoNative_Potato
```

**Review by VSadov — APPROVED** [review]
LGTM Thank you!!

File: src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/CMakeLists.txt [hunk 1]
```diff
@@ -73,6 +73,17 @@ if (GEN_SHARED_LIB)
     target_link_libraries(System.Security.Cryptography.Native.Apple
         ${NATIVE_LIBS_EXTRA}
     )
+
+    if (NOT CLR_CMAKE_TARGET_MACCATALYST AND NOT CLR_CMAKE_TARGET_IOS AND NOT CLR_CMAKE_TARGET_TVOS)
+        add_custom_command(TARGET System.Security.Cryptography.Native.Apple POST_BUILD
+            COMMENT "Verifying System.Security.Cryptography.Native.Apple points against entrypoints.c "
+            COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../verify-entrypoints.sh
+                $<TARGET_FILE:System.Security.Cryptography.Native.Apple>
+                ${CMAKE_CURRENT_SOURCE_DIR}/../../Unix/System.Security.Cryptography.Native.Apple/entrypoints.c
```
Comment (jkotas) [inline src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/CMakeLists.txt:82]:
```suggestion
                ${CMAKE_CURRENT_SOURCE_DIR}/entrypoints.c
```

Comment (vcsjones) [reply src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/CMakeLists.txt]: 🤦 thanks!

**Pull request merged**